harvester/DBInterface.py

```diff
@@ -77,107 +69,199 @@ def _prep(self, statement):
 		return statement
 
 
-	def create_repo(self, repository_url, repository_name, repository_type, repository_thumbnail="", item_url_pattern=""):
+	def update_repo(self,repo_id,repo_url,repo_set,repo_name,repo_type,enabled,repo_thumbnail,item_url_pattern,abort_after_numerrors,max_records_updated_per_run,update_log_after_numitems,record_refresh_days,repo_refresh_days):
+		con = self.getConnection()
+		records=[]
+		with con:
+			if self.dbtype == "sqlite":
+				con.row_factory = self.getRow()
+				cur = con.cursor()
+			if self.dbtype == "postgres":
+				from psycopg2.extras import RealDictCursor
+				cur = con.cursor(cursor_factory = RealDictCursor)
+			if repo_id > 0:
+				# Existing repo
+				try:
+					self.logger.debug("This repo already exists in the database; updating")
+					cur.execute(self._prep("""UPDATE repositories 
+						set repository_url=?, repository_set=?, repository_name=?, repository_type=?, repository_thumbnail=?, last_crawl_timestamp=?, item_url_pattern=?,enabled=?,
+						abort_after_numerrors=?,max_records_updated_per_run=?,update_log_after_numitems=?,record_refresh_days=?,repo_refresh_days=?
+						WHERE repository_id=?"""), (repo_url, repo_set, repo_name, repo_type, repo_thumbnail, time.time(), item_url_pattern, enabled, 
+						abort_after_numerrors,max_records_updated_per_run,update_log_after_numitems,record_refresh_days,repo_refresh_days,repo_id) )
+				except self.dblayer.IntegrityError:
+					# record already present in repo
+					return repo_id
+			else:
+				# Create new repo record
+				try:
+					self.logger.debug("This repo did not exist in the database; adding")
+					if self.dbtype == "postgres":
```

Comment:
Is it necessary to split this logic by postgres/SQLite? Lite should support the fetchone() method...